src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -860,8 +856,8 @@ class TaskBuilderTest extends MarathonSpec {
     // The taskName is the elements of the path, reversed, and joined by dots
     assert("frontend.product" == taskInfo.getName)
     assert(2 == taskPorts.size)
-    assert(taskPorts.head == range.get.getBegin.toInt)
-    assert(taskPorts(1) == range.get.getEnd.toInt)
+    assert(taskPorts.contains(range.get.getBegin.toInt))
+    assert(taskPorts.contains(range.get.getEnd.toInt))
```

Comment:
The original test checked that the task got assigned exactly the first and last ports of the offered range.

This is often, but not always the case. I think that it is more appropriate to check only that all the ports it got are contained in the offer.